R/draw_majorityline.R

```diff
@@ -0,0 +1,76 @@
+#' Draw majority line
+#' @param n The number of seats required for a majority
+#' @param label A logical variable for labelling majority threshold. Defaults to TRUE.
+#' @param type type of parliament (horseshoe, semicircle,opposing benches)
+#' @examples
+#' data <- ggparliament::election_data %>% filter(year == "2016" & country == "USA" & house == "Representatives")
+#' usa_data <- parliament_data(election_data = data, type = "semicircle", party_seats = data$seats, party_names = data$party_short, parl_rows = 8, total_seats = sum(data$seats))
+#' ggplot(usa_data, aes(x, y, color=party_long)) + geom_parliament_seats() + draw_majorityline(n = 316, type = 'opposing_benches', label = FALSE)
+#' @author Zoe Meers
+#' @export
+
+
+draw_majoritythreshold <- function(...,
+                                   n = NULL,
+                                   label = TRUE,
+                                   type = c(
+                                     "horseshoe",
+                                     "semicircle",
+                                     "opposing_benches"
+                                   )) {
+  structure(
+    list(
+      type = type,
+      n = n,
+      label = label
+    ),
+    class = "majorityLine"
+  )
+}
+
+
+ggplot_add.majorityLine <- function(object, plot, object_name) {
+  new_dat <- plot$data %>%
+    dplyr::filter(government == 1) %>%
+    dplyr::filter(dplyr::row_number() == object$n)
+  x_pos <- new_dat$x
+  y_pos_oppbenches <- new_dat$y
+
+
+
+
+  if (object$type == "semicircle") {
```

Comment:
see draw_legislaturelabels. Will probably simplify the ifelse chains using switch calls where possible. Will pull this across before I do that because I don't want to spend ages doing small edits here and there when the branch works but think there's a lot of scope to reduce some of the functions.

Is there a way to pipe the type through ggplot (e.g. p <- ggplot(data = df, aes(type = type)) + draw_majoritythreshold() ) or does it need to be defined in every extra geom/scale/... ?